gui/app/simple_clock.cpp

```diff
@@ -0,0 +1,61 @@
+#include "simple_clock.h"
+#include <QTime>
+
+SimpleClock::SimpleClock(QObject *parent)
+    : QObject(parent), enabled_(true), update_interval_(1000), hour_(0),
+      minute_(0), second_(0), timer_(this) {
+
+  connect(&timer_, &QTimer::timeout, this, &SimpleClock::update_timer);
+  timer_.start();
+}
+
+SimpleClock::~SimpleClock() { timer_.stop(); }
+
+bool SimpleClock::GetEnabled() const { return enabled_; }
+
+int SimpleClock::GetUpdateInterval() const { return update_interval_; }
+
+int SimpleClock::GetHour() const { return hour_; }
+
+int SimpleClock::GetMinute() const { return minute_; }
+
+int SimpleClock::GetSecond() const { return second_; }
+
+void SimpleClock::SetUpdateInterval(int update_interval) {
+  if (update_interval_ != update_interval) {
+    update_interval_ = update_interval;
+    emit updateIntervalChanged();
+    update_timer();
+  }
+}
+
+void SimpleClock::SetEnabled(bool enabled) {
+  if (enabled_ != enabled) {
+    enabled_ = enabled;
+    emit enabledChanged();
+    update_timer();
+  }
+}
+
+void SimpleClock::update_timer() {
+  auto time = QTime::currentTime();
+
+  hour_ = time.hour();
+  emit hourChanged();
```

Comment:
Can we just store the whole QTime value and emit a signal that time changed, rather than having accessors and signals for hour/minute/second?

Reply:
`QTime` does not inherit `QObject`, so it would be moved to QML side as a `QVariant`. That means we will not have IDE auto-complete.

For C++ code, autocomplete is not exactly necessary, but writing QML code is fundamental since the amount of elements is properties are overwhelming to memorize and there is no easy shortcut around moving from QML to the C++ counterpart and vice-versa.

I would recommend to keep it as it is for now.

Reply:
I'm suggesting to keep the GetHour,Minute,Second functions, but simply store the QTime as a member variable, and have a single signal that notifies about all three of these. Then the current function would look like:

time_ = QTime::currentTime();
emit timeChanged();

but the caller would still access the components of the time via GetHour,Minute,Second.